tests/update_reference_data.sh

```diff
@@ -246,40 +58,38 @@ test -z "$FAILED_TESTS" && exit 5
 
 for failed_test in $FAILED_TESTS
 do
-  failed=`echo $failed_test | sed -e 's/.*:compareECLFiles_//g'`
-  for test_name in ${!tests[*]}
-  do
-    binary=`echo ${tests[$test_name]} | awk -F ' ' '{print $1}'`
-    dirname=`echo ${tests[$test_name]} | awk -F ' ' '{print $2}'`
-    casename=`echo ${tests[$test_name]} | awk -F ' ' '{print $3}'`
-    if grep -q "$failed" <<< "$binary+$casename"
-    then
-      copyToReferenceDir \
+  grep -q compareECLFiles <<< $failed_test
+  test $? -ne 0 && continue
+  failed_test=`echo $failed_test | sed -e 's/.*://g'`
+  # Extract test properties
+  binary=`cat $BUILD_DIR/CTestTestfile.cmake | grep set_tests_properties\($failed_test| awk -F ' ' '{for (i=0; i <= NF; ++i) { if ($i == "SIMULATOR") {print $(i+1); exit}}}' | sed -e 's/\"//g'`
+  dir_name=`cat $BUILD_DIR/CTestTestfile.cmake | grep set_tests_properties\($failed_test| awk -F ' ' '{for (i=0; i <= NF; ++i) { if ($i == "DIRNAME") {print $(i+1); exit}}}' | sed -e 's/\"//g'`
+  file_name=`cat $BUILD_DIR/CTestTestfile.cmake | grep set_tests_properties\($failed_test| awk -F ' ' '{for (i=0; i <= NF; ++i) { if ($i == "FILENAME") {print $(i+1); exit}}}' | sed -e 's/\"//g'`
+  test_name=`cat $BUILD_DIR/CTestTestfile.cmake | grep set_tests_properties\($failed_test| awk -F ' ' '{for (i=0; i <= NF; ++i) { if ($i == "TESTNAME") {print $(i+1); exit}}}' | sed -e 's/\"//g'`
```

Comment:
You can reduce the number of processes here by writing a slightly more targeted awk script leveraging awk's built-in pattern matching engine:
```sh
v=$(awk -v search="set_tests_properties\\\($failed_test" \
  '$1 ~ search { for (i = 3; i <= NF; ++i) { if ($i == "SOMETHING") { val = $(i + 1); gsub(/"/, "", val); print val; exit } } } \
  $BUILD_DIR/CTestTestfile.cmake)
```
Note that I'm starting at `i = 3`, because `$0` is the entire line, `$1` would be the `set_tests_properties` command, and `$2` would be the word `PROPERTIES`.  If you don't want to make the latter two assumptions you should nevertheless start at `i = 1`.

Building on that we could even have a separate awk script in a file named `getprop.awk` of the form
```awk
$1 ~ search {
    for (i = 3; i <= NF; ++i) {
        if ($i == prop) {
            val = $(i + 1)
            gsub(/"/, "", val)
            print val
            exit
        }
    }
}
```
and invoke this from `update_reference_data.sh` as
```sh
binary=$(awk -v search="set_tests_properties\\\($failed_test" -v prop="SIMULATOR" -F getprop.awk $BUILD_DIR/CTestTestfile.cmake)
dir_name=$(awk -v search="set_tests_properties\\\($failed_test" -v prop="DIRNAME" -F getprop.awk $BUILD_DIR/CTestTestfile.cmake)
file_name=$(awk -v search="set_tests_properties\\\($failed_test" -v prop="FILENAME" -F getprop.awk $BUILD_DIR/CTestTestfile.cmake)
test_name=$(awk -v search="set_tests_properties\\\($failed_test" -v prop="TESTNAME" -F getprop.awk $BUILD_DIR/CTestTestfile.cmake)
```